### ISSUE TITLE
Ability to link local packages without running pnpm install

### ISSUE THREAD

**Issue by tgroutars** [pnpm/pnpm#4674]
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))
Happy to help building this if you can point me to the relevant parts in the codebase :)

**Describe the user story**

As a developer, I want to be able to relink the local packages without running `pnpm install` after I made changes to a package.

Example:
I have 2 packages, package A and package B. Package A depends on package B
```
A -> B
```

Let's say I create a new file `B/src/new-file.ts` in B and rebuild it using `pnpm build`. I have this new file under `B/dist/new-file.js`. 

Since B is hard linked into A's node_modules, the file is not added there and is not available inside A. `A/node_modules/B/dist/new-file.js` does not exist.

If I have a `postinstall` script in B, I can run `pnpm install` and the package will be linked after the build has finished, which is great. But that means I have to run `pnpm install` everytime this happens. 

I'd like to be able to run a `pnpm` command that just links B again inside A's node_modules, without running a full install.

This would also solve another issue that we have at the moment because we're implementing turborepo in our monorepo. 

We have postinstall scripts in all our libraries. We want to make use of turborepo to do this so we can cache our builds. The problem is, we can only run turborepo from the root.

One solution that we've thought about is to have `bootstrap` commands in each library that needs to be built, and a global `postinstall` script in the root. 

```
"scripts": {
  "postinstall": "turbo run bootstrap"
}
```

The problem with this approach is that we sometimes only install the depedencies of one packages and its dependencies 
```
pnpm install --filter=some-package...
```

Using that approach, the global `postinstall` script would try to run `bootstrap` in every package, even the ones where we didn't install dependencies, because we don't have information about the filters that were passed to `pnpm install`, so this is no good.

The other solution, more tedious but acceptable, is to require to run `pnpm bootstrap` manually after running `pnpm install`. This way we can pass the filters depending on the context. The problem is that this step will create the build folder in those packages, but they won't appear in their dependents node_modules, so we need a way to relink the packages without running `pnpm install` again

**Describe the solution you'd like**

A `pnpm` command that only links local packages without running `pnpm install`
For example
```
pnpm link-local-packages
```

Not fan of the name so if you find a better one, I'm open to it 😅 

**Describe the drawbacks of your solution**

Can't think of any issues

**Describe alternatives you've considered**

Another solution that would be super neat would be to add an argument to `pnpm run` to automatically relink the package where the command is being run *after* the command has run.

So we could for example run `pnpm run --link build`, and pnpm would automatically relink the package once the build is done

**Comment by zkochan** [comment]
I don't have objections.

Linking currently happens after the lifecycle scripts run: https://github.com/pnpm/pnpm/blob/bf6839a55643327f0a70f0ef5b3ea605d65ee6fe/packages/lifecycle/src/runLifecycleHooksConcurrently.ts#L55-L82

Maybe that code should be moved to a separate package. And a new commands plugin should be created. Not sure about the name of the command. `bit link` actually does something similar. But in npm/pnpm/yarn link already has some other purpose.

**Comment by benyap** [comment]
@tgroutars this describes the exact issue we currently have with using `pnpm` too. Would love to see this feature get implemented!
